src/nodes/primitive/range.ts

```diff
@@ -63,17 +91,24 @@ export type Range = [Bound] | [Bound<MinComparator>, Bound<MaxComparator>]
 export interface RangeNode extends BaseNode<{ kind: "range"; rule: Range }> {
     min: Bound<MinComparator> | undefined
     max: Bound<MaxComparator> | undefined
+    numericMin: Bound<MinComparator, number | Date> | undefined
+    numericMax: Bound<MaxComparator, number | Date> | undefined
+    value: number | Date
```

Comment:
This cannot be called value, as it looks like it is the first bound limit?

If you think about what would make sense for the API based on how it's being used, maybe you're looking for something like `rangeKind: "numeric" | "date"`